src/Explorer/Controls/Settings/SettingsUtils.tsx

```diff
@@ -0,0 +1,331 @@
+import * as React from "react";
+import * as AutoPilotUtils from "../../../Utils/AutoPilotUtils";
+import * as SharedConstants from "../../../Shared/Constants";
+import * as CommonConstants from "../../../Common/Constants";
+import * as DataModels from "../../../Contracts/DataModels";
+import { AutopilotTier } from "../../../Contracts/DataModels";
+import {
+  computeAutoscaleUsagePriceHourly,
+  getPriceCurrency,
+  getCurrencySign,
+  getAutoscalePricePerRu,
+  getMultimasterMultiplier,
+  computeRUUsagePriceHourly,
+  getPricePerRu,
+  calculateEstimateNumber
+} from "../../../Utils/PricingUtils";
+
+/*
+Class used to track baseline and current values of a state object. If both the values differ,
+isDirty is set to true. The isDirty value is used to change css of the component and to govern logic
+around whether the current value should be persistently saved or not.
+*/
+export type StatefulValuesType = boolean | string | number | DataModels.IndexingPolicy;
+
+export interface StatefulValue<StatefulValuesType> {
+  baseline: StatefulValuesType;
+  current: StatefulValuesType;
+  isValid: boolean;
+}
+
+const getStringValue = (value: StatefulValuesType): string => {
+  const type = typeof value;
+  switch (type) {
+    case "string":
+    case "undefined":
+    case "number":
+    case "boolean":
+      return value?.toString();
+
+    default:
+      return JSON.stringify(value);
+  }
+};
+
+export function isDirty(value: StatefulValue<StatefulValuesType>): boolean {
+  const current = getStringValue(value.current);
+  const baseline = getStringValue(value.baseline);
+
+  if (current !== baseline) {
+    return true;
+  }
+
+  return false;
+}
+
+export function getAutoPilotV3SpendElement(
```

Comment:
These functions below are a typical good use case for a `React.FunctionComponent`. Make this a separate component. We can then unit test with `maxAutoPiloThroughputSet` set or not set, etc.
A react component should be reusable, so it might make more sense to group them into one or only a couple of components rather than converting every single one of them to a component.